File: fixtures/scale/javascript/large/src/core/codec.js
function toHex(text) {
  return Array.from(text)
    .map((ch) => (ch.charCodeAt(0) % 256).toString(16).padStart(2, '0'))
    .join('');
}

function fingerprint(text) {
  let total = 0;
  for (const ch of text) total += ch.charCodeAt(0);
  return (total % 65536).toString(16).padStart(4, '0');
}

export default { toHex, fingerprint };

File: fixtures/scale/lua/large/core/codec.lua
local function to_hex(text)
  return (text:gsub(".", function(ch) return string.format("%02x", ch:byte()) end))
end

local function fingerprint(text)
  local total = 0
  for i = 1, #text do total = total + text:byte(i) end
  return string.format("%04x", total % 65536)
end

return { to_hex = to_hex, fingerprint = fingerprint }
